src/NServiceBus.Core/Receiving/ProcessingCompleted.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Event raised when the transport has completed processing a message.
+    /// </summary>
+    public class ProcessingCompleted
+    {
+        /// <summary>
+        /// Constructs the event.
+        /// </summary>
+        public ProcessingCompleted(string messageId, bool wasAcknowledged, Dictionary<string, string> headers, DateTimeOffset startedAt, DateTimeOffset completedAt)
+        {
+            MessageId = messageId;
+            WasAcknowledged = wasAcknowledged;
+            Headers = headers;
+            StartedAt = startedAt;
+            CompletedAt = completedAt;
+        }
+
+        /// <summary>
+        /// The ID of the message.
+        /// </summary>
+        public string MessageId;
```

Comment:
I see that `IncomingMessage`, which is a property on `ReceivePipelineCompleted`, has both `MessageId` and `NativeMessageId`. Which one is this?

Reply:
This is the native message id. I will rename

Reply:
Renamed